### PR TITLE
Capabilities Cache

### PR DESCRIPTION
**Pull request by anthonydresser**
Cleans up how we handle cacheing capabilities for services and isolates it all to the capabilities service. Removes a lot of unnecessary code, while also fixing the bug that the connection wouldn't be buffered properly.

### CODE REVIEW

File: src/sql/parts/connection/common/connectionConfig.ts [hunk 1]
```diff
@@ -29,32 +29,14 @@ export interface ISaveGroupResult {
...
-		this._providerCapabilitiesMap = {};
-		this.setCachedMetadata(cachedMetadata);
-		if (this._capabilitiesService && this._capabilitiesService.onCapabilitiesReady()) {
-			this._capabilitiesService.onCapabilitiesReady().then(() => {
-				this.setCachedMetadata(this._capabilitiesService.getCapabilities());
-			});
-		}
-	}
-
-	public setCachedMetadata(cachedMetadata: sqlops.DataProtocolServerCapabilities[]): void {
```
Comment (kburtram) [inline src/sql/parts/connection/common/connectionConfig.ts]: I think @saurabh500 added these caches to help speed up startup.  Do you know if removing these will regress startup experience?

**Comment by saurabh500** [comment]
Cache was added because initial connection dialog was waiting for the services to come up to query capabilities. 
If this is removed then ops studio will startup and then no connection dialog box will popup until all the services are up.

File: src/sql/parts/connection/common/connectionProfile.ts [hunk 2]
```diff
@@ -170,8 +170,19 @@ export class ConnectionProfile extends ProviderConnectionInfo implements interfa
 		};
 	}
 
-	public static createFromStoredProfile(profile: interfaces.IConnectionProfileStore, serverCapabilities: sqlops.DataProtocolServerCapabilities): ConnectionProfile {
-		let connectionInfo = new ConnectionProfile(serverCapabilities, undefined);
+	public static fromIConnectionProfile(capabilitiesService: ICapabilitiesService, profile: interfaces.IConnectionProfile) {
```
Comment (kburtram) [inline src/sql/parts/connection/common/connectionProfile.ts]: should this method have a return type?

**Comment by kburtram** [comment]
Thanks @saurabh500! I setup a meeting on Monday to discuss if we want Connection Dialog to show automatically on 1st launch (among other 1st launch issues).  Assuming we do, then I think we'll need to fix that experience.

**Comment by anthonydresser** [comment]
@saurabh500 as far as I understand, that functionality was in the protocol layer. It would pull the capabilities from the config.json instead of sending the request to the client. I removed that awhile ago as part of the 3.0 upgrade with the promise I would add it back in. I had a branch where I did that but I was waiting on a lot of my other changes to go in. This change is part of that larger change.

The caches I removed were simply caching the data that the capabilitiesService was getting, which it was getting from sending the request to the extension. So the caches I removed were basically just isolated caches. This changes just takes all the different places we were cacheing capabilities and isolates it all to the capabilitiesService itself.

File: build/lib/i18n.js [hunk 3]
```diff
@@ -533,12 +531,9 @@ function getResource(sourceFile) {
     else if (/^vs\/workbench/.test(sourceFile)) {
         return { name: 'vs/workbench', project: workbenchProject };
     }
-
-    // {{SQL CARBON EDIT}}
```
Comment (AbbiePetcht) [inline build/lib/i18n.js]: >// {{SQL CARBON EDIT}} [](start = 4, length = 22)

why did you remove this line?

Comment (anthonydresser) [reply build/lib/i18n.js]: this is a compiled file. It compiled and got rid of it

**Comment by saurabh500** [comment]
I synced up with @anthonydresser offline and his improvements are different from the work that I had done. 
He will be working on the initial capabilities caching on first startup, by moving the cached capabilities to package.json so that on the first start up, the capabilities are available for different UI elements. 

Thanks for explaining this. The approach LGTM.
